### PR TITLE
f

### PR DESCRIPTION
**Pull request by ClarenceGardner**

### CODE REVIEW

File: o4/o4.py [hunk 1]
```diff
@@ -1062,7 +1067,7 @@ def o4_status(changelist, depot, check_all, quick):
     print(f"*** INFO: o4 status {here}")
     if not os.path.isdir('.o4'):
         print("Never synced with o4.")
-        return
+        return 1
```
Comment (philipbergen) [inline o4/o4.py]: This could be argued both 0 and 1.

File: o4/o4.py [hunk 2]
```diff
@@ -1166,8 +1171,8 @@ def o4_status(changelist, depot, check_all, quick):
         print("")
         print(f"*** INFO: Besides the {len(has_open)} file{s} opened for edit, "
               f"all files passed the checksum test.")
-
-    return True
+        return 0
+    return len(all_fnames)
```
Comment (philipbergen) [inline o4/o4.py]: Should be length of naughty I think. There's no error just because you have files open for edit.

File: o4/o4.py [hunk 3]
```diff
@@ -1052,6 +1052,11 @@ def man(cmd):
 
 
 def o4_status(changelist, depot, check_all, quick):
+    '''
+    Checks for files that don't match Perforce.
+    Returns the number of files that are bad in any way,
```
Comment (philipbergen) [inline o4/o4.py]: Returns the number of files, excluding files opened for edit/add/delete, that are not matching checksum.

File: o4/o4.py [hunk 4]
```diff
@@ -1570,7 +1575,7 @@ def main():
             if opts['--report']:
                 print(opts['--report'].format(**locals()))
         if opts['status']:
-            o4_status(opts['@'], opts['<path>'], opts['-f'], opts['-q'])
+            ec = o4_status(opts['@'], opts['<path>'], opts['-f'], opts['-q'])
```
Comment (philipbergen) [inline o4/o4.py]: That becomes weird if you have exit code 672890.

**Comment by rohanrebello** [comment]
Test

File: o4/o4.py (hunk 4 above)
Comment (ClarenceGardner) [reply o4/o4.py]: good point

**Pull request closed without merging**